core/tee/tee_svc_cryp.c

```diff
@@ -2192,7 +2192,14 @@ tee_svc_obj_ed25519_parse_params(const TEE_Attribute *params, size_t num_params,
 	for (n = 0; n < num_params; n++) {
 		switch (params[n].attributeID) {
 		case TEE_ATTR_EDDSA_PREHASH:
-			*ph_flag = true;
+			if (params[n].content.value.b)
+				return TEE_ERROR_BAD_PARAMETERS;
+			if (!params[n].content.value.a)
+				*ph_flag = false;
+			else if (params[n].content.value.a == 1)
+				*ph_flag = true;
+			else
+				return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
I suppose this should ideally have been introduced in commit 0aaad418ac8b ("core: crypto: add Ed25519 support") and would justify a separate commit.

Looks OK with the spec and with xtest since commit [7596bd335890](https://github.com/OP-TEE/optee_test/commit/7596bd335890) ("regression 4016: fix TEE_ATTR_EDDSA_PREHASH parameter") but CC @varder in case this could cause compatibility troubles.

Reply:
OK, done in https://github.com/OP-TEE/optee_os/pull/5707